pkg/reconciler/v1alpha1/taskrun/resources/cloudevent/cloud_event_controller.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudevent
+
+import (
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// InitializeCloudEvents initializes the CloudEvents part of the
+// TaskRunStatus from a slice of PipelineResources
+func InitializeCloudEvents(tr *v1alpha1.TaskRun, prs []*v1alpha1.PipelineResource) {
+	// FIXME(afrittoli) If there are no events this is run every time
```

Comment:
is there a github issue we could put in this FIXME as well? I'm also not quite sure what the implication is - does it mean the code is hit every time (which seems okay?) or does something actually happen that we don't want to have happen?

Reply:
In case no cloud events are defined, the loop in L34-39 will be executed on every reconcile - not a big deal really :)